sympy/concrete/summations.py

```diff
@@ -1530,66 +1521,70 @@ def get_residue_factor(numer, denom, alternating):
     if denom.degree(i) - numer.degree(i) < 2:
         return None
 
-    if (a, b) == (S.NegativeInfinity, S.Infinity):
-        poles = get_poles(denom)
-        if poles is None:
-            return None
-        int_roots, nonint_roots = poles
-
-        if int_roots:
-            return None
-
-        residue_factor = get_residue_factor(numer, denom, alternating)
-        residues = [residue(residue_factor, z, root) for root in nonint_roots]
-        return -S.Pi * sum(residues)
+    # handle non-even function with limits (-oo, oo), (-oo, k) or (k, oo)
 
-    if not (a.is_finite and b is S.Infinity):
-        return None
+    both_inf = (a,b) == (S.NegativeInfinity, S.Infinity)
 
     if not is_even_function(numer, denom):
         # Try shifting summation and check if the summand can be made
-        # and even function from the origin.
+        # even wrt the origin.
         # Sum(f(n), (n, a, b)) => Sum(f(n + s), (n, a - s, b - s))
         shift = get_shift(denom)
 
-        if not shift.is_Integer:
-            return None
-        if shift == 0:
+        if not shift:
             return None
 
-        numer = numer.shift(shift)
-        denom = denom.shift(shift)
+        if shift.is_Integer:
+            numer = numer.shift(shift)
+            denom = denom.shift(shift)
+            if not is_even_function(numer, denom):
+                return None
+
+            if alternating:
+                f = S.NegativeOne**i * (S.NegativeOne**shift * numer.as_expr() / denom.as_expr())
+            else:
+                f = numer.as_expr() / denom.as_expr()
+            return eval_sum_residue(f, (i, a-shift, b-shift))
+        elif not both_inf:
+            return None  # can't make a shift that keeps limits integer
 
-        if not is_even_function(numer, denom):
+    if both_inf:  # even or not
```

Comment:
I think that the both_inf case can be handled above before checking `is_even_function` now that the residue calculation is fixed.